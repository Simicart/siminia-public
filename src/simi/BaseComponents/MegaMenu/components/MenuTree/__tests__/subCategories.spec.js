import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import SubCategories from '../subCategories';

jest.mock('../../MenuTree/menuItem', () => 'MenuItem');

test('renders without data', () => {
  const tree = createTestInstance(<SubCategories />);
  expect(tree.toJSON()).toMatchSnapshot();
});

const items = [
  {
    name: 'Bottoms',
    id: 'category-node-11',
    url: 'venia-bottoms.html',
    parent_id: null,
    __typename: 'MenuCategoryItem'
  },
  {
    name: 'Pants & Shorts',
    id: 'category-node-12',
    url: 'venia-bottoms/venia-pants.html',
    parent_id: 'category-node-11'
  }
];

test('renders correct tree', () => {
  const tree = createTestInstance(<SubCategories items={items} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
