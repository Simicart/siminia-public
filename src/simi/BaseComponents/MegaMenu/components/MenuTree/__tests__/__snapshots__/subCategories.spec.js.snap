// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct tree 1`] = `
<ul
  className="  "
  style={
    Object {
      "--am-mega-menu-columns-count": 4,
    }
  }
>
  <MenuItem
    __typename="MenuCategoryItem"
    id="category-node-11"
    level={1}
    name="Bottoms"
    parent_id={null}
    url="venia-bottoms.html"
  />
  <MenuItem
    id="category-node-12"
    level={1}
    name="Pants & Shorts"
    parent_id="category-node-11"
    url="venia-bottoms/venia-pants.html"
  />
</ul>
`;

exports[`renders without data 1`] = `null`;
